### PR TITLE
Add support for menu controls

### PR DESCRIPTION
**Pull request by otaku42**
I've implemented some kind of support for menu-style controls (V4L2_CTRL_TYPE_MENU), allowing to access the menu items defined for such control. 

I suggest to add this support to v4l2py. But I'm not entirely confident if my approach is "the right way to do it", especially since I'm completely new to V4L2, so I'd welcome your review and will happily accept any suggestions for modifications. One thing I'm uncertain about is the use of copy.deepcopy() when yielding menu items - I assume this may be a good idea, but maybe it's not needed?

Usage example:
```
import v4l2py

cam = v4l2py.Device("/dev/video4", True)
if cam is not None:
    cam.open()
    print(f"*** {cam.info.card} ***")
    for c in cam.controls.values():
        print("0x%08x:" % c.id, c)
        for item in c.menu.values():
            print(" - %s" % item)
    cam.close()
else:
    print("Uh oh, something went wrong")
```

Sample output:
```
ignored unknown pixel format M420 (808596557)
*** Microsoft® LifeCam Studio(TM): ***
0x00980900: <Control name=Brightness, type=INTEGER, min=30, max=255, step=1>
0x00980901: <Control name=Contrast, type=INTEGER, min=0, max=10, step=1>
0x00980902: <Control name=Saturation, type=INTEGER, min=0, max=200, step=1>
0x0098090c: <Control name=White Balance, Automatic, type=BOOLEAN, min=0, max=1, step=1>
0x00980918: <Control name=Power Line Frequency, type=MENU, min=0, max=2, step=1>
 - <MenuItem index=0 name=Disabled>
 - <MenuItem index=1 name=50 Hz>
 - <MenuItem index=2 name=60 Hz>
0x0098091a: <Control name=White Balance Temperature, type=INTEGER, min=2500, max=10000, step=1>
0x0098091b: <Control name=Sharpness, type=INTEGER, min=0, max=50, step=1>
0x0098091c: <Control name=Backlight Compensation, type=INTEGER, min=0, max=10, step=1>
0x009a0901: <Control name=Auto Exposure, type=MENU, min=0, max=3, step=1>
 - <MenuItem index=1 name=Manual Mode>
 - <MenuItem index=3 name=Aperture Priority Mode>
0x009a0902: <Control name=Exposure Time, Absolute, type=INTEGER, min=1, max=10000, step=1>
0x009a0908: <Control name=Pan, Absolute, type=INTEGER, min=-529200, max=529200, step=3600>
0x009a0909: <Control name=Tilt, Absolute, type=INTEGER, min=-432000, max=432000, step=3600>
0x009a090a: <Control name=Focus, Absolute, type=INTEGER, min=0, max=40, step=1>
0x009a090c: <Control name=Focus, Automatic Continuous, type=BOOLEAN, min=0, max=1, step=1>
0x009a090d: <Control name=Zoom, Absolute, type=INTEGER, min=0, max=317, step=1>
```

And that's the output of v4l2-ctl for the same device:
```
$ v4l2-ctl -d /dev/video4 -L

User Controls

                     brightness 0x00980900 (int)    : min=30 max=255 step=1 default=133 value=45
                       contrast 0x00980901 (int)    : min=0 max=10 step=1 default=5 value=3
                     saturation 0x00980902 (int)    : min=0 max=200 step=1 default=103 value=0
        white_balance_automatic 0x0098090c (bool)   : default=1 value=1
           power_line_frequency 0x00980918 (menu)   : min=0 max=2 default=2 value=2
                                0: Disabled
                                1: 50 Hz
                                2: 60 Hz
      white_balance_temperature 0x0098091a (int)    : min=2500 max=10000 step=1 default=4500 value=4500 flags=inactive
                      sharpness 0x0098091b (int)    : min=0 max=50 step=1 default=25 value=25
         backlight_compensation 0x0098091c (int)    : min=0 max=10 step=1 default=0 value=0

Camera Controls

                  auto_exposure 0x009a0901 (menu)   : min=0 max=3 default=3 value=3
                                1: Manual Mode
                                3: Aperture Priority Mode
         exposure_time_absolute 0x009a0902 (int)    : min=1 max=10000 step=1 default=156 value=156 flags=inactive
                   pan_absolute 0x009a0908 (int)    : min=-529200 max=529200 step=3600 default=0 value=0
                  tilt_absolute 0x009a0909 (int)    : min=-432000 max=432000 step=3600 default=0 value=0
                 focus_absolute 0x009a090a (int)    : min=0 max=40 step=1 default=0 value=0 flags=inactive
     focus_automatic_continuous 0x009a090c (bool)   : default=1 value=1
                  zoom_absolute 0x009a090d (int)    : min=0 max=317 step=1 default=0 value=0
```

### CODE REVIEW

**Review by tiagocoutinho — APPROVED** [review]
Looks good to me!
Thanks for the improvement

**Comment by tiagocoutinho** [comment]
[Flake8 test didn't pass](
https://github.com/tiagocoutinho/v4l2py/actions/runs/4251225340/jobs/7393300606#step:5:55)
Would you mind fixing it before the merge?
Thanks in advance

File: v4l2py/device.py [hunk 1]
```diff
@@ -289,6 +292,17 @@ def iter_read_controls(fd):
...
 
 
+def iter_read_menu(fd, ctrl):
+    menu = raw.v4l2_querymenu()
+    menu.id = ctrl.id
+    for menu in iter_read(
+            fd, IOC.QUERYMENU, menu,
+            start=ctrl.info.minimum, stop=ctrl.info.maximum+1,
+            step=ctrl.info.step, ignore_einval=True
+        ):
```
Comment (tiagocoutinho) [inline v4l2py/device.py:302]: I think this is what makes flake8 complain

File: v4l2py/device.py [hunk 2]
```diff
@@ -754,13 +768,29 @@ def write(self, data: bytes) -> None:
...
+
+
 class Control:
     def __init__(self, device, info):
         self.device = device
         self.info = info
         self.id = self.info.id
         self.name = info.name.decode()
         self.type = ControlType(self.info.type)
+        if self.type == raw.V4L2_CTRL_TYPE_MENU:
```
Comment (tiagocoutinho) [inline v4l2py/device.py:788]: Maybe you could use `ControlType`?
Something like:
```if self.type == ControlType.MENU:```

**Review by tiagocoutinho — CHANGES_REQUESTED** [review]
Some minor comments
+ 
needs a flake8 fix

File: v4l2py/device.py (hunk 2 above)
Comment (otaku42) [reply v4l2py/device.py]: Good point, makes sense. I've changed it as suggested.

**Review by tiagocoutinho — APPROVED** [review]
Awesome 🤩 
Merging now! Thanks again

**Review by tiagocoutinho — APPROVED** [review]
Awesome 🤩 
Merging now! Thanks again

**Pull request merged**